pkg/api/scheme/scheme.go

```diff
@@ -29,11 +31,29 @@ import (
 	storagev1 "k8s.io/api/storage/v1"
 	storagev1beta1 "k8s.io/api/storage/v1beta1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
 	apiregistrationv1beta1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1beta1"
 )
 
-var Scheme *runtime.Scheme
+type FilteredScheme struct {
+	*runtime.Scheme
+	excludedGVKs map[schema.GroupVersionKind]bool
+}
+
+func (f *FilteredScheme) New(kind schema.GroupVersionKind) (runtime.Object, error) {
+	if f.excludedGVKs[kind] {
+		return nil, fmt.Errorf("%s is explicitly excluded from this scheme", kind.String())
```

Comment:
Are there any side effects of returning an error here? Will the call be retried because an error is returned here?

Reply:
With how we use it we expect an error when something is not found.  I am just returning a more specific error incase someone is debugging.